packages/amazonq/src/inlineChat/provider/inlineChatProvider.ts

```diff
@@ -0,0 +1,209 @@
+        triggerID: string
+    ) {
+        const triggerEvent = this.triggerEventsStorage.getTriggerEvent(triggerID)
+        if (triggerEvent === undefined) {
+            return
+        }
+
+        if (triggerEvent.tabID === 'no-available-tabs') {
+            return
+        }
+
+        if (triggerEvent.tabID === undefined) {
+            setTimeout(() => {
+                this.generateResponse(triggerPayload, triggerID).catch((e) => {
+                    getLogger().error('generateResponse failed: %s', (e as Error).message)
+                })
+            }, 20)
+            return
+        }
+
+        const tabID = triggerEvent.tabID
+
+        const credentialsState = await AuthUtil.instance.getChatAuthState()
+
+        if (
+            !(credentialsState.codewhispererChat === 'connected' && credentialsState.codewhispererCore === 'connected')
+        ) {
+            await this.sendAuthNeededExceptionMessage(credentialsState, tabID, triggerID)
+            return
+        }
+        triggerPayload.useRelevantDocuments = false
+
+        const request = triggerPayloadToChatRequest(triggerPayload)
+        const session = this.sessionStorage.getSession(tabID)
+        getLogger().info(
+            `request from tab: ${tabID} conversationID: ${session.sessionIdentifier} request: ${JSON.stringify(
+                request
+            )}`
+        )
+
+        let response: GenerateAssistantResponseCommandOutput | undefined = undefined
+        session.createNewTokenSource()
+        try {
+            response = await session.chat(request)
+            getLogger().info(
+                `response to tab: ${tabID} conversationID: ${session.sessionIdentifier} requestID: ${response.$metadata.requestId} metadata: ${JSON.stringify(
+                    response.$metadata
+                )}`
+            )
+        } catch (e: any) {
+            this.processException(e, tabID)
+            this.errorEmitter.fire()
+            return undefined
+        }
+
+        return response
+    }
+
+    private processException(e: any, tabID: string) {
+        let errorMessage = ''
+        let requestID = undefined
+        const defaultMessage = 'Failed to get response'
+        if (typeof e === 'string') {
+            errorMessage = e.toUpperCase()
+        } else if (e instanceof SyntaxError) {
+            // Workaround to handle case when LB returns web-page with error and our client doesn't return proper exception
+            errorMessage = AwsClientResponseError.tryExtractReasonFromSyntaxError(e) ?? defaultMessage
+        } else if (e instanceof CodeWhispererStreamingServiceException) {
+            errorMessage = e.message
+            requestID = e.$metadata.requestId
+        } else if (e instanceof Error) {
+            errorMessage = e.message
+        }
+
+        this.errorEmitter.fire()
+        void vscode.window.showErrorMessage(errorMessage)
```

Comment:
If code is called by a command, none of this is needed. Just throw the error, and our common handler will display it.

If throwing an error is not acceptable from some inline chat context for some reason, then at least look at the common error handling code and think about how to use it. It shouldn't be re-invented here.

Reply:
Kept what logic is needed here, moved to using `ToolkitError`